DataDog/datadog-operator
# Use same instance label value for network policies

### What does this PR do?

When creating the agent components, we set the instance label as `<dda-name>-<component-suffix>`. We need to use that as the pod selector when creating network policies or else they won't be targeting the correct pods.

### Motivation

https://github.com/DataDog/datadog-operator/issues/1490

### Additional Notes

Anything else we should know when reviewing?

### Minimum Agent Versions

Are there minimum versions of the Datadog Agent and/or Cluster Agent required?

* Agent: vX.Y.Z
* Cluster Agent: vX.Y.Z

### Describe your test plan

* Create a DDA with network policies enabled:
```yaml
spec:
  global:
    networkPolicy:
      create: true
      flavor: kubernetes
```
* Check that the network policy pod selector uses the same instance label value that's on the corresponding pod. Example:
```
$ kubectl get networkpolicy
datadog-agent-agent                   app.kubernetes.io/instance=datadog-agent-agent,app.kubernetes.io/part-of=default-datadog--agent                   7h56m
```
The label key and value `app.kubernetes.io/instance=datadog-agent-agent` from the network policy should match the agent pod's instance label key and value. Double check that the part of labels match as well:
```
$ kubectl describe pod <agent-pod>
Labels:           agent.datadoghq.com/component=agent
                  [...]
                  app.kubernetes.io/instance=datadog-agent-agent
                  [...]
                  app.kubernetes.io/part-of=default-datadog--agent
```


### Checklist
